deploy/chart/templates/plugin.yaml

```
@@ -1,17 +1,20 @@
-{{- if .Values.plugin.enabled -}}
+{{- range .Values.nodepools -}}
+{{- if $.Values.plugin.enabled }}
+{{- $copy := deepCopy .csi }}
+{{- $csi := mergeOverwrite $.Values.csi $copy }}
```

```suggestion
{{- range $key, $nodepool := .Values.nodepools -}}
{{- if and $.Values.plugin.enabled (ne $nodepool nil) }}
{{- $nodepool := merge $nodepool $.Values }}
```
* I want to support configs other than `.csi`, e.g. cloud nodes may use VPC registry URL, but edge node must use Internet URL.
* Allow user to set a node pool to nil to disable it.
* Merge values into nodepool, so that we don't need deep copy.

---

It‘s a useful advice to merge values into nodepool directly so that we can rewrite any field of original values.
> Allow user to set a node pool to nil to disable it.
I'm confused about this suggestion. If users want to disable a nodepool, just delete it from nodepools, why setting it to nil?

---

Because it is straightforward to write `helm install --set nodepools.default=nil`. But not clear how to delete it from CLI.